frontend/src/scenes/surveys/SurveyEdit.tsx

```diff
@@ -79,11 +88,31 @@ export default function SurveyEdit(): JSX.Element {
         surveysActionsAvailable,
     } = useValues(surveysLogic)
     const { featureFlags } = useValues(enabledFeaturesLogic)
+    const [visible, setVisible] = useState(false)
     const sortedItemIds = survey.questions.map((_, idx) => idx.toString())
     const { thankYouMessageDescriptionContentType = null } = survey.appearance ?? {}
     const surveysRecurringScheduleDisabledReason = surveysRecurringScheduleAvailable
         ? undefined
         : 'Upgrade your plan to use repeating surveys'
+    const surveysAdaptiveLimitsDisabledReason = surveysRecurringScheduleAvailable
+        ? undefined
+        : 'Upgrade your plan to use an adaptive limit on survey responses'
+
+    useMemo(() => {
+        if (survey.responses_limit && survey.responses_limit > 0) {
+            setDataCollectionType('until_limit')
+        } else if (
+            survey.response_sampling_interval &&
+            survey.response_sampling_interval > 0 &&
+            survey.response_sampling_interval_type !== '' &&
+            survey.response_sampling_limit &&
+            survey.response_sampling_limit > 0
+        ) {
```

Comment:
This logic isn't intuitive to me at first glance. Would it be worth adding a comment describing what it's supposed to do?

Reply:
Good idea, doing that now!